ic-os/defs.bzl

```diff
@@ -192,29 +183,25 @@ def icos_build(
                 for k, v in (
                     image_deps["bootfs"].items() + [
                         (version_txt, "/version.txt:0644"),
-                    ] + ([(extra_boot_args, "/extra_boot_args:0644")] if "boot_args_template" not in image_deps else []) +
-                    ([(boot_args, "/boot_args:0644")] if "boot_args_template" in image_deps else [])
+                        (boot_args, "/boot_args:0644"),
+                        (extra_boot_args, "/extra_boot_args:0644"),
+                    ]
                 )
             },
             tags = ["manual", "no-cache"],
         )
 
-        # The kernel command line (boot args) was previously split into two parts:
-        # 1. Dynamic args calculated at boot time in grub.cfg
-        # 2. Static args stored in EXTRA_BOOT_ARGS on the boot partition
+        # The kernel command line (boot args) is generated from boot_args_template:
+        # - For OS requiring root signing: Template includes ROOT_HASH placeholder that gets substituted with dm-verity hash
+        # - For OS not requiring root signing: Template is used as-is without ROOT_HASH substitution
         #
-        # For stable and predictable measurements with AMD SEV, we now pre-calculate and combine both parts
-        # into a single complete kernel command line that is:
-        # - Generated during image build
-        # - Stored statically on the boot partition
-        # - Measured as part of the SEV launch measurement
-        #
-        # For HostOS and SetupOS, we continue
```

Comment:
Should we keep the old comment around until we really remove the old path? The original comment also had:

> For backwards compatibility in the GuestOS and compatibility with the HostOS and SetupOS, we continue
> to support the old way of calculating the dynamic args (see :extra_boot_args) and we derive boot_args
> from it.

And maybe should say:
```
For backwards compatibility in GuestOS and HostOS, we continue to support the old way of calculating the dynamic args (see :extra_boot_args).
```